### PR TITLE
Improve error handling in quay-permissions

### PR DESCRIPTION
**Pull request by Piojo**
There are more methods that can raise exceptions, so we cover each
iteration (i.e, each reviewed repository) in a try-except block.

Also, we replace the old `RequestsException` with the `HTTPError`
class supplied by `requests`. This gives us more information,
including access to the failed response and request.

This way, we can get complete error messages, such as:

```[2021-09-06 22:12:54] [ERROR] [quay_permissions.py:run:116] - Could not manage repo permissions on repo $REPO_NAME,team $TEAM_NAME - status code: 403 - resource URL: https://quay.io/api/v1/repository/$REPOSITORY_SHA/$REPOSITORY_NAME/permissions/team/$TEAM_NAME
Traceback (most recent call last):
[...]
requests.exceptions.HTTPError: 403 Client Error: FORBIDDEN for url: ...
```

### CODE REVIEW

File: reconcile/utils/quay_api.py [hunk 1]
```diff
@@ -124,9 +116,7 @@ def create_or_update_team(self, team: str, role="member",
             payload.update({'description': description})
 
         r = requests.put(url, headers=self.auth_header, json=payload)
-
-        if not r.ok:
-            raise RequestsException(r)
+        r.raise_for_status()
```
Comment (rporres) [inline reconcile/utils/quay_api.py:119]: Please update docstring to reflect this change

File: reconcile/utils/quay_api.py [hunk 2]
```diff
@@ -97,8 +90,7 @@ def add_user_to_team(self, user, team):
         url = "{}/organization/{}/team/{}/members/{}".format(
             self.api_url, self.organization, team, user)
         r = requests.put(url, headers=self.auth_header)
-        if not r.ok:
-            raise RequestsException(r)
+        r.raise_for_status()
```
Comment (rporres) [inline reconcile/utils/quay_api.py:93]: can you add a docstring describing this?

File: reconcile/utils/quay_api.py [hunk 3]
```diff
@@ -43,8 +37,7 @@ def list_team_members(self, team, **kwargs):
                 f"org {self.organization}. "
                 f"contact org owner to create the "
                 f"team manually.")
-        elif not r.ok:
-            raise RequestsException(r)
+        r.raise_for_status()
```
Comment (rporres) [inline reconcile/utils/quay_api.py:40]: Please update docstring to reflect this change

File: reconcile/quay_permissions.py [hunk 4]
```diff
@@ -77,48 +78,54 @@ def run(dry_run):
...
-                        current_role = \
-                            quay_api.get_repo_team_permissions(
-                                repo_name, team_name)
-                        if current_role != role:
-                            logging.info(
-                                ['update_role', org_key, repo_name,
-                                 team_name, role])
-                            if not dry_run:
-                                try:
+                    try:
```
Comment (rporres) [inline reconcile/quay_permissions.py:81]: This `try` block is either navigating through dicts or doing calls. I don't get understand need for such a broad scope. Could we just start it in `get_repo_team_permissions`?

File: reconcile/quay_permissions.py [hunk 5]
```diff
@@ -77,48 +78,54 @@ def run(dry_run):
...
-                                        'details: {%s}',
-                                        repo_name, team_name, e
-                                    )
+                    except HTTPError:
+                        error = True
+                        logging.exception(
+                            f'Could not manage repo permissions on {repo_name}'
+                            f' - team {team_name}'
+                        )
+                    except Exception:
```
Comment (rporres) [inline reconcile/quay_permissions.py:122]: I see no different treatment in the catch-all or in the `HTTPError`. Why the difference? If we reduced the scope of the `try` block, could we narrow the exceptions here?

Comment (Piojo) [reply reconcile/quay_permissions.py:122]: It started with somewhat different messages, and then converged, sorry, I'll simplify this.

File: reconcile/quay_permissions.py (hunk 4 above)
Comment (Piojo) [reply reconcile/quay_permissions.py:81]: The goal here is to do as much work as possible even if some repositories or teams fail.

If we can't even get repo permissions we can't continue the work of the iteration. Just having `try: except: log;continue` is more and harder-to-read code. Just having the happy path inside the `try` block leads to less code, and makes sure any other mistake I may have forgotten to consider is handled. This is the top-level in which we want unexpected exceptions handled and reported grafecully.

File: reconcile/quay_permissions.py [hunk 6]
```diff
@@ -86,39 +86,38 @@ def run(dry_run):
                             continue
 
                         perm_org_key = (
-                          permission['quayOrg']['instance']['name'],
-                          permission['quayOrg']['name']
+                            permission['quayOrg']['instance']['name'],
+                            permission['quayOrg']['name']
```
Comment (rporres) [inline reconcile/quay_permissions.py:90]: please avoid unrelated linting changes.

File: reconcile/quay_permissions.py [hunk 7]
```diff
@@ -86,39 +86,38 @@ def run(dry_run):
...
                         )
 
                         if perm_org_key != org_key:
-                            logging.warning('wrong org, should be %s', org_key)
+                            logging.warning(
+                                f'wrong org, should be {org_key}'
+                            )
```
Comment (rporres) [inline reconcile/quay_permissions.py:96]: There are endless discussions about avoiding interpolation in logging or not... In this case it should not matter but I'm all for avoiding unrelated changes in a PR 😂

**Review by rporres — APPROVED** [review]
a couple of nits, nothing important

**Comment by Piojo** [comment]
I hadn't pushed my fixes to your docstring requests. Sorry!  They are in now.
